Tests/scripts/utils/get_modified_files_for_testing.py

```diff
@@ -0,0 +1,214 @@
+            py_to_be_removed.add(file_path)
+
+    # remove python files
+    if py_to_be_removed:
+        types_to_files[FileType.PYTHON_FILE] = types_to_files[FileType.PYTHON_FILE] - py_to_be_removed
+
+    return types_to_files
+
+
+def create_type_to_file(files_string: str) -> Dict[FileType, Set[str]]:
+    """Classifies the files in the diff list (files_string) using tools.find_type
+
+    Returns:
+        A dict of {FileType: Set of files}
+    """
+    types_to_files: Dict[FileType, Set[str]] = dict()
+    for line in files_string.split("\n"):
+        if line:
+            file_status, file_path = line.split(maxsplit=1)
+            file_status = file_status.lower()
+            # Get to right file_path on renamed
+            if file_status.startswith("r"):
+                _, file_path = file_path.split(maxsplit=1)
+            file_status = file_status.lower()
+            # ignoring deleted files.
+            # also, ignore files in ".circle", ".github" and ".hooks" directories and .
+            if file_path:
+                if (
+                    file_status in ("m", "a") or file_status.startswith("r")
+                ) and not file_path.startswith("."):
+                    file_type = tools.find_type(file_path) or resolve_type(file_path)
+                    if file_type in types_to_files:
+                        types_to_files[file_type].add(file_path)
+                    elif file_type is not None:
+                        types_to_files[file_type] = {file_path}
+
+    types_to_files = remove_python_files(types_to_files)
+
+    return types_to_files
+
+
+def get_modified_files_for_testing(
+    git_diff: str,
+) -> Tuple[List[str], List[str], List[str], bool, List[str], set, bool, bool]:
+    """
+    Gets git diff string and filters those files into tests:
+
+    Args:
+        git_diff: a git diff output (with --name-only flag)
+    Returns:
+        modified_files: Modified YMLs for testing (Integrations, Scripts, Playbooks).
+        modified_tests: Test playbooks.
+        changed_common: Globally used YMLs (Like CommonServerPython).
+        is_conf_json: If Tests/Conf.json has been changed.
+        sample_tests: Files to test, Like the infrastructures files.
+        modified_metadata: Pack names of changed metadata files.
+        is_reputations_json: If any reputation file changed.
+        is_indicator_json: If any indicator file changed.
+    """
+    sample_tests: Set[str] = set()  # Files to test, Like the infrastructures files.
+    modified_metadata: Set[str] = set()  # Metadata files.
+    modified_files: Set[str] = set()  # Modified YMLs for testing (Integrations, Scripts, Playbooks).
+    types_to_files: Dict[FileType, Set[str]] = create_type_to_file(git_diff)  # Mapping of the files FileType: file path
+
+    # Checks if any common file represents in it
+    file_to_check_common = types_to_files.get(
+        FileType.INTEGRATION, set()).union(types_to_files.get(FileType.SCRIPT, set()))
```

Comment:
Why would we classify any of the commons as `FileType.INTEGRATION`? Looks like all the commons are scripts: `["scripts/script-CommonIntegration.yml", "scripts/script-CommonIntegrationPython.yml",
                   "Packs/Base/Scripts/script-CommonServer.yml", "scripts/script-CommonServerUserPython.yml",
                   "scripts/script-CommonUserServer.yml",
                   "Packs/Base/Scripts/CommonServerPython/CommonServerPython.yml"]`